src/cm_stats_api.h

```diff
@@ -70,6 +70,15 @@ stats_ns_t *
 stats_ns_t *
   stats_register_ns(stats_recorder_t *, stats_ns_t *, const char *);
 
+/* Deregister and free a namespace.
+ * Results are undefined if ns is still being used during or after this call
+ * if the return value is true.
+ * This function deregisters ns from parent within rec. If successful, ns is
+ * freed.
+ */
```

Comment:
It would mean we have to expose the free call making the internal mechanics a bit more accessible.  Given that you cannot be writing to this namespace when you deregister it (unsafe)... I think you could just merge the namespace into a new one then deregister w/ free as it is.

I'm not sure there is much advantage to giving that control.  If you feel you need it, implement it and glue it into this PR.